src/org/thoughtcrime/securesms/PinnedMessagesListActivity.java

```diff
@@ -0,0 +1,156 @@
+    }
+
+    protected void initializeActionBar() {
+        ActionBar supportActionBar = getSupportActionBar();
+        if (supportActionBar == null) throw new AssertionError();
+
+        supportActionBar.setDisplayHomeAsUpEnabled(false);
+        supportActionBar.setCustomView(R.layout.conversation_title_view);
+        supportActionBar.setDisplayShowCustomEnabled(true);
+        supportActionBar.setDisplayShowTitleEnabled(false);
+    }
+
+
+    private void initializeViews() {
+        ActionBar supportActionBar = getSupportActionBar();
+        if (supportActionBar == null) throw new AssertionError();
+
+        titleView = (ConversationTitleView) supportActionBar.getCustomView();
+        titleView.setOnBackClickedListener(view -> super.onBackPressed());
+    }
+
+    private void initializeResources() {
+        if (recipient != null) recipient.removeListener(this);
+
+        recipient = Recipient.from(this, getIntent().getParcelableExtra(ADDRESS_EXTRA), true);
+        threadId = getIntent().getLongExtra(THREAD_ID_EXTRA, -1);
+        archived = getIntent().getBooleanExtra(IS_ARCHIVED_EXTRA, false);
+        distributionType = getIntent().getIntExtra(DISTRIBUTION_TYPE_EXTRA, ThreadDatabase.DistributionTypes.DEFAULT);
+        GlideRequests glideRequests = GlideApp.with(this);
+
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.JELLY_BEAN) {
+            LinearLayout conversationContainer = ViewUtil.findById(this, R.id.conversation_container);
+            conversationContainer.setClipChildren(true);
+            conversationContainer.setClipToPadding(true);
+        }
+
+        setActionBarColor(recipient.getColor());
+
+        titleView.setCustomeTitle(glideRequests, recipient,"Pinned Messages", "With " + recipient.getProfileName());
+
+        recipient.addListener(this);
+    }
+
+    private void setActionBarColor(MaterialColor color) {
+        ActionBar supportActionBar = getSupportActionBar();
+        if (supportActionBar == null) throw new AssertionError();
+        supportActionBar.setBackgroundDrawable(new ColorDrawable(color.toActionBarColor(this)));
+        setStatusBarColor(color.toStatusBarColor(this));
+    }
+
+    @Override
+    public void setThreadId(long threadId) {
+        this.threadId = threadId;
+    }
+
+    @Override
+    public void onPointerCaptureChanged(boolean hasCapture) {
+
+    }
+
+    @Override
+    public void onModified(Recipient recipient) {
+        Log.w(TAG, "onModified(" + recipient.getAddress().serialize() + ")");
+        Util.runOnMain(() -> {
+            Log.w(TAG, "onModifiedRun(): " + recipient.getRegistered());
+            setActionBarColor(recipient.getColor());
+            invalidateOptionsMenu();
+        });
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        super.onOptionsItemSelected(item);
+        switch (item.getItemId()) {
+            case android.R.id.home:
+                finish();
+                return true;
```

Comment:
Not a big issue but this return true is redundant if you were to let it fall through.

Reply:
Its to make it easy to add more in front or after but I can change it.

Reply:
thumbs up to the first part of your comment, thumbs down to the second. Keep it how it is if you didn't already change it. Better to avoid bugs.

Reply:
Might